opwen_email_server/utils/email_parser.py

```diff
@@ -80,3 +83,28 @@ def _get_recipients(email: dict) -> Iterable[str]:
 def get_domains(email: dict) -> Iterable[str]:
     return frozenset(address.split('@')[-1]
                      for address in _get_recipients(email))
+
+
+def _get_as_base64(image_url):
+    res = requests.get(image_url)
+
+    if res.status_code != 200:
+        res.raise_for_status()
+
+    img_type = res.headers['Content-Type']
+    img_content = b64encode(res.content).decode('ascii')
+    base64 = "data:" + img_type + ";base64," + img_content
+
+    return base64
+
+
+def convert_img_url_to_base64(email: dict) -> dict:
+    email_body = email.get('body', '')
+    if email_body:
+        soup = BeautifulSoup(email_body, 'html.parser')
+        for img in soup.find_all('img'):
+            img_url = img.get('src')
+            img_base64 = _get_as_base64(img_url)
+            img['src'] = img_base64
+        email['body'] = str(soup)
```

Comment:
Given that the method returns a dictionary, it would be unexpected to mutate the input dictionary. Please make a copy of the email with `dict(email)` and mutate/return that copy.